sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -125,26 +125,34 @@ export const imdsMsi: MSI = {
         // returned quickly from the endpoint, proving its availability.
         const request = createPipelineRequest(requestOptions);
 
-        // Default to 300 if the default of 0 is used.
+        // Default to 1000 if the default of 0 is used.
         // Negative values can still be used to disable the timeout.
-        request.timeout = options.requestOptions?.timeout || 300;
+        request.timeout = options.requestOptions?.timeout || 1000;
 
         // This MSI uses the imdsEndpoint to get the token, which only uses http://
         request.allowInsecureConnection = true;
-
+        let response;
         try {
           logger.info(`${msiName}: Pinging the Azure IMDS endpoint`);
-          await identityClient.sendRequest(request);
+          response = await identityClient.sendRequest(request);
         } catch (err: unknown) {
           // If the request failed, or Node.js was unable to establish a connection,
           // or the host was down, we'll assume the IMDS endpoint isn't available.
           if (isError(err)) {
             logger.verbose(`${msiName}: Caught error ${err.name}: ${err.message}`);
           }
+          // This is a special case for Docker Desktop which responds with a 403 with a message that contains "A socket operation was attempted to an unreachable network"
+          // rather than just timing out, as expected.
           logger.info(`${msiName}: The Azure IMDS endpoint is unavailable`);
           return false;
         }
-
+        if ((response as any).status === 403) {
```

Comment:
Can we declare `response` as `PipelineResponse` above so we can remove the `as any` bits?

Reply:
oh yes